frontend/server/src/Controllers/Run.php

```diff
@@ -915,6 +921,144 @@ public static function apiDetails(\OmegaUp\Request $r): array {
             $response['judged_by'] = strval($run->judged_by);
         }
 
+        $problemArtifacts = new \OmegaUp\ProblemArtifacts(
+            $problem->alias,
+            $run->commit
+        );
+
+        $cases = self::getCases(
+            $problemArtifacts,
+            'cases',
+            $problem->alias,
+            $run->commit
+        );
+        /** @var int $responseSize */
+        $responseSize = array_reduce(
+            $cases,
+            /**
+             * @param array{mode: int, type: string, id: string, size: int, path: string} $case
+             */
+            function (?int $sum, $case): int {
+                if (is_null($sum)) {
+                    return $case['size'];
+                }
+                $sum += $case['size'];
+                return $sum;
+            }
+        );
+
+        if (
+            $problem->show_diff === \OmegaUp\ProblemParams::NO_SHOW_DIFFS
+            || $responseSize > 4096 // Forcing to hide diffs when inputs/outpus exceed 4kb
+        ) {
+            $response['show_diff'] = \OmegaUp\ProblemParams::NO_SHOW_DIFFS;
+            return $response;
+        }
+        if ($problem->show_diff === \OmegaUp\ProblemParams::SHOW_ALL_DIFFS) {
+            $dataCases = self::getProblemCases(
+                $problemArtifacts,
+                'cases',
+                $problem->alias,
+                $run->version
+            );
+            $response['cases'] = $dataCases;
+            $response['show_diff'] = \OmegaUp\ProblemParams::SHOW_ALL_DIFFS;
+            return $response;
+        }
+        $dataCases = self::getProblemCases(
+            $problemArtifacts,
+            'examples',
+            $problem->alias,
+            $run->version
+        );
+        $response['cases'] = $dataCases;
+        $response['show_diff'] = \OmegaUp\ProblemParams::SHOW_ONLY_EXAMPLE_DIFF;
+
+        return $response;
+    }
+
+    /**
+     * @return ProblemCases
+     */
+    private static function getProblemCases(
+        \OmegaUp\ProblemArtifacts $problemArtifacts,
+        string $directory,
+        string $problemAlias,
+        ?string $version
+    ): array {
+        return \OmegaUp\Cache::getFromCacheOrSet(
+            \OmegaUp\Cache::DATA_CASES,
+            "{$problemAlias}-{$version}-{$directory}",
+            /** @return ProblemCases */
+            function () use (
+                $problemArtifacts,
+                $directory,
+                $problemAlias,
+                $version
+            ) {
+                return self::getProblemCasesImpl(
+                    $problemArtifacts,
+                    $directory,
+                    $problemAlias,
+                    $version
+                );
+            },
+            24 * 60 * 60 /*expire in 1 day*/
+        );
+    }
+
+    /**
+     * @return list<array{mode: int, type: string, id: string, size: int, path: string}>
+     */
+    private static function getCases(
```

Comment:
esto no obtiene los casos, sino la metadata, entonces qué tal

```suggestion
    private static function getProblemCasesMetadata(
```

? y en vez de `DATA_CASES_FILES`, se puede usar `PROBLEM_CASES_METADATA`.

Reply:
Listo